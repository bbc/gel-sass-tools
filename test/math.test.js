describe('math()', () => {
  describe('with positive values', () => {
    it('should return a quarter of the given value', async () => {
      const css = await global.compileCSS('test/data/math/quarter.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return a third of the given value', async () => {
      const css = await global.compileCSS('test/data/math/third.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return a halve of the given value', async () => {
      const css = await global.compileCSS('test/data/math/halve.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return two times the given value', async () => {
      const css = await global.compileCSS('test/data/math/double.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return three times the given value', async () => {
      const css = await global.compileCSS('test/data/math/triple.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return four times the given value', async () => {
      const css = await global.compileCSS('test/data/math/quadruple.scss');

      expect(css).toMatchSnapshot();
    });
  });

  describe('with negative values', () => {
    it('should return a quarter of the given value', async () => {
      const css = await global.compileCSS('test/data/math/quarter-negative.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return a third of the given value', async () => {
      const css = await global.compileCSS('test/data/math/third-negative.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return a halve of the given value', async () => {
      const css = await global.compileCSS('test/data/math/halve-negative.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return two times the given value', async () => {
      const css = await global.compileCSS('test/data/math/double-negative.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return three times the given value', async () => {
      const css = await global.compileCSS('test/data/math/triple-negative.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return four times the given value', async () => {
      const css = await global.compileCSS('test/data/math/quadruple-negative.scss');

      expect(css).toMatchSnapshot();
    });
  });
});
