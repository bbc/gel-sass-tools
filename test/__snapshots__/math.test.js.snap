// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`math() with negative values should return a halve of the given value 1`] = `
".demo {
  padding: -5px;
}"
`;

exports[`math() with negative values should return a quarter of the given value 1`] = `
".demo {
  padding: -3px;
}"
`;

exports[`math() with negative values should return a third of the given value 1`] = `
".demo {
  padding: -3px;
}"
`;

exports[`math() with negative values should return four times the given value 1`] = `
".demo {
  padding: -40px;
}"
`;

exports[`math() with negative values should return three times the given value 1`] = `
".demo {
  padding: -30px;
}"
`;

exports[`math() with negative values should return two times the given value 1`] = `
".demo {
  padding: -20px;
}"
`;

exports[`math() with positive values should return a halve of the given value 1`] = `
".demo {
  padding: 5px;
}"
`;

exports[`math() with positive values should return a quarter of the given value 1`] = `
".demo {
  padding: 3px;
}"
`;

exports[`math() with positive values should return a third of the given value 1`] = `
".demo {
  padding: 3px;
}"
`;

exports[`math() with positive values should return four times the given value 1`] = `
".demo {
  padding: 40px;
}"
`;

exports[`math() with positive values should return three times the given value 1`] = `
".demo {
  padding: 30px;
}"
`;

exports[`math() with positive values should return two times the given value 1`] = `
".demo {
  padding: 20px;
}"
`;
